pytoda/datasets/tests/test_polymer_dataset.py

```diff
@@ -348,5 +408,112 @@ def test___getitem___with_annotation_column_names(self) -> None:
+
+    @mock_input
+    def test___getitem__(self, mock_file_1, mock_file_2) -> None:
+        polymer_dataset = _PolymerDatasetNoAnnotation(
+            smi_filepaths=[mock_file_1.filename, mock_file_2.filename],
+            entity_names=['monomer', 'cATalysT'],
+            remove_bonddir=True
+        )
+        (
+            pad_ind, monomer_start_ind, monomer_stop_ind, catalyst_start_ind,
+            catalyst_stop_ind, c_ind, o_ind, n_ind, s_ind
+        ) = _getitem_helper(polymer_dataset)
+
+        # test retrieving one sample of each entity
+        monomer = polymer_dataset['monomer', 0]
+        catalyst = polymer_dataset['catalyst', 1]
+
+        self.assertEqual(
+            monomer.numpy().flatten().tolist(), [
+                pad_ind, monomer_start_ind, c_ind, c_ind, o_ind,
+                monomer_stop_ind
+            ]
+        )
+
+        self.assertEqual(
+            catalyst.numpy().flatten().tolist(), [
+                *(29 * [pad_ind]), catalyst_start_ind, c_ind, c_ind,
+                catalyst_stop_ind
+            ]
+        )
+
+    @mock_input
+    def test__return_modes(self, mock_file_1, mock_file_2) -> None:
+        polymer_dataset = _PolymerDatasetNoAnnotation(
+            smi_filepaths=[mock_file_1.filename, mock_file_2.filename],
+            entity_names=['monomer', 'cATalysT'],
+            remove_bonddir=True
+        )
+        polymer_dataset.set_mode_smiles()
+        monomer = polymer_dataset['monomer', 3]
+
+        self.assertEqual(monomer, '<MONOMER_START>NCCS<MONOMER_STOP>')
+
+        polymer_dataset.set_mode_tensor()
+        monomer = polymer_dataset['monomer', 3]
+
+        (
+            pad_ind, monomer_start_ind, monomer_stop_ind, catalyst_start_ind,
+            catalyst_stop_ind, c_ind, o_ind, n_ind, s_ind
+        ) = _getitem_helper(polymer_dataset)
+
+        self.assertEqual(
+            monomer.numpy().flatten().tolist(), [
+                monomer_start_ind, n_ind, c_ind, c_ind, s_ind,
+                monomer_stop_ind
+            ]
+        )
+
+
+class TestPolymerDataset(unittest.TestCase):
+
+    @mock_input
+    def test___init__annotation(self, mock_file_1, mock_file_2) -> None:
+        annotated_content = os.linesep.join(
+            [
+                'label_0,label_1,monomer,catalyst',
+                '2.3,3.4,CHEMBL545,CHEMBL17',
+                '4.5,5.6,CHEMBL17564,CHEMBL6402',  # yapf: disable
+                '6.7,7.8,CHEMBL602,CHEMBL6402',
+                '6.7,7.8,CHEMBL54556,CHEMBL5434'
+            ]
+        )
+        with TestFileContent(annotated_content) as annotation_file:
+            PolymerDataset(
+                smi_filepaths=[mock_file_1.filename, mock_file_2.filename],
+                entity_names=['monomer', 'cATalysT'],
+                annotations_filepath=annotation_file.filename
+            )
+
+    @mock_input
+    def test___init__no_annotation(self, mock_file_1, mock_file_2) -> None:
+        PolymerDataset(
```

Comment:
- It’s fine to only test the initialisation of the polymerDataset method but we’re overlooking in the tests the case of not passing any smiles language AND not passing any smiles transform (like removeBonddir). In that case, no pass is performed over the .smi files and effectively, no tokens are added to the smiles/polymer language which makes each get item call failing (unless `_returning_tensors=False`). We should definitely add a test for this case.
 `SMILESDataset` works like this in order to support a fast initialisation if you bring along a predefined smiles language. Maybe the fastest way to fix this is to add a `parse_data` flag to `SMILESDataset` that defaults to True. Because here, I think that there is no way around pre-defining the PolymerLanguage object before creating the SMILESDataset instances (otherwise the language is not shared).

Reply:
Yeah, I definitively should add some extra tests. I have it in mind. I am actually still making changes on the fly on the datasets, so for sure it's not yet ready for merge (I should have put a note on the original message of the PR). In any case, I'll fix this.